### ISSUE TITLE
Broken dark themes offered in settings

### ISSUE THREAD

**Issue by ParadoxKG** [gitextensions/gitextensions#10490]
### Environment

- Git Extensions 4.0.0.15569
- Build cba315df5366fca01240d93b6e5a5077fd5e6630
- Git 2.38.1.windows.1
- Microsoft Windows NT 10.0.19044.0
- .NET 6.0.11
- DPI 120dpi (125% scaling)

### Issue description

Dark themes are known to be broken in GITExtension 4, but they are still offered in the settings.

Expected: Broken elements are not offered or are not selectable in the settings.

### Steps to reproduce

- Open Settings/Appearance/Colors
- Enable Theme dark or darksilver

### Did this work in previous version of GitExtensions?

Not needed as themes were working.

**Comment by vbjay** [comment]
https://github.com/gitextensions/gitextensions/releases/tag/v4

Next time read release notes.

**Comment by pmiossec** [comment]
Sorry @vbjay but the assessment of @ParadoxKG is good I think.
If I'm not wrong, GE setup is still providing the `dark.css` and `darksilver.css` files.
And because GE 4 doesn't support dark theme, I think we should remove them form the setup otherwise users will select it.

It's a first step in the direction but, if we won't have dark theme anytime soon, maybe also we should prevent loading dark theme inside the application and displaying an error message...

**Comment by vbjay** [comment]
I thought it even popped a message of saying not valid and switching to light theme on first run.

**Comment by ParadoxKG** [comment]
> Next time read release notes.

Actually, that leads me to another issue. I had not particular plan to update GE, but I was prompted by GE 3.X "There is a new version. Do you want to update?". If you confirm that dialog, you never come by the relase notes and you are never informed that the new version is broken - at least from the perspective of a user who has everything else in dark mode it is a significant problem.

Can I downgrade from GE4 back to the latest GE3 or will that cause problems/loss of settings?

**Comment by gerhardol** [comment]
> Can I downgrade from GE4 back to the latest GE3 or will that cause problems/loss of settings?

Should be OK, there is some minor migrations in the settings to 4.0 for PuTTY and I believe themes.

**Comment by ParadoxKG** [comment]
Thank you. Reverted back to GE3, seems to work fine.

**Comment by lucas-foreflight** [comment]
Agreed with @ParadoxKG 
I was holding off with this update waiting for patches to prevent the firstmover effect, only after I went through the trouble of manually updating and installing putty do I find out the dark theme is gone.. Which is a pretty big deal breaker. 
Functionally 3.X was great, the loss of dark theme IMO makes 4.X worse even if it bears performance improvements

**Comment by pmiossec** [comment]
> Agreed with @ParadoxKG I was holding off with this update waiting for patches to prevent the firstmover effect, only after I went through the trouble of manually updating and installing putty do I find out the dark theme is gone.. Which is a pretty big deal breaker. Functionally 3.X was great, the loss of dark theme IMO makes 4.X worse even if it bears performance improvements

@lucas-foreflight I don't know what to do with this type of comments
The team have put a lot of efforts in trying to keep this feature but the feature is no more working due to a breaking change made in the dotnet runtime: https://github.com/gitextensions/gitextensions/issues/9191#issuecomment-1312595006

The v4 deliver a lot of value in other parts of the application so I think me made the good decision to release it.
What do you think we should have done? Wait before doing a release until the dotnet framework provide a workaround? At the moment, there is no deadline provided so it was not accepatble...

**Comment by lucas-foreflight** [comment]
@pmiossec I apologize if I came off rude. I dont mean to disrespect the efforts your team puts into this tool. 
It comes down to what he stated, that if it was more obvious that one breaking change would be the removal of the dark theme, then I personally would have held off updating too. Ofc it should not get in the way of developing efforts. Now I can fix it on my part and simply revert to 3.X

**Comment by ParadoxKG** [comment]
I don't know about the history and technical background of theming. If there is a potential workaround/alternative implementation for the theming and you needed to weigh the effort for going that way against the importance of theming, then such feedback would indicate that it is a showstopper for many users and it might be worthwhile to address it.

If it is a non-negotiable thing and you have no options except waiting for Microsoft to fix it, the insight is not really helpful, of course. :-)
If the problem is being tracked somewhere for .net and could be upvoted, it might be helpful to provide a link to there for interested parties like me.
